### DISCUSSION TITLE
Blogpost 4

### DISCUSSION THREAD

**Discussion by ChristophNie** [LeonRother/SE_Platzverwaltung#5] [General]
# Modellierung der Klassen

In der letzten Woche haben wir uns auf die Kernklassen unseres Projekts konzentriert. Unser Hauptaugenmerk lag darauf, eine stabile Grundlage zu schaffen, bevor wir uns weiteren Aspekten widmen. Geplant ist außerdem die Einführung einer zusätzlichen Klasse, die für den Datenbankzugriff zuständig sein wird. Die genaue Ausgestaltung dieser Klasse steht jedoch noch nicht fest, daher haben wir sie vorerst außen vor gelassen. In dem docs Ordner sind die Klassendiagramme zu finden.

Die Beziehungen zwischen den Klassen haben wir bisher ignoriert, da wir uns zunächst auf die Implementierung der grundlegenden Funktionen konzentrieren wollten. Das gibt uns die Freiheit, die Struktur ohne unnötige Komplexität zu entwickeln.

# SOLID-Prinzip

Bezüglich der SOLID-Prinzipien haben wir uns entschieden, das Single Responsibility Principle (SRP) zu berücksichtigen. Jede Klasse soll eine klar definierte Aufgabe übernehmen, um die Wartbarkeit und Flexibilität unseres Codes zu verbessern.

# Aktualisierung alter Dokumente

Gibt es vorerst keine, werden aber bei Bedarf aktualisiert.

**Comment by LemChenju** [comment]
Hey,

Ich würde versuchen mein Feedback so kurz wie möglich zu halten.:

### Verbesserungsvorschläge
- Könnt Ihr ein Bild eurer Klassendiagrammes reinstellen und nicht nur die Drawio-xmls?
- Im Diagramm fehlt dem Passwort ein Datentyp (selbes gilt für Passwort im GlobalAdmin Diagramm)
- Die MitarbeiterService hat einen schreibfehler im Namen

### Fragen
- Ich finde es gut das Ihr das SRP schonmal im Rahmen des Klassendiagrammes berücksichtigt, plant Ihr im Rahmen eurer Erweiterung durch die Klasse für die Interaktion mit der Datenbank weitere SOLID-Prinzipien zu integrieren?

GaLiGrü 😊
Paula, Safae, Samuel, Valentin, Jonas

**Reply by Hpixpoke** [reply]
Danke für's Feedback. Ich habe ein paar Änderungen vorgenommen:
- Tippfehler behoben
- Bilder hochgeladen
- Passwort-Typen eingefügt (es steht Hash drin, gemeint ist im DB-Kontext natürlich ein String)

Zu weiteren Prinzipien: ist erstmal nichts geplant, weil uns das SRP schon etwas ratlos lässt, was unser API-Design angeht. Kann sich in Zukunft aber natürlich noch ändern ;)
